packages/table/src/components/table/table.tsx

```
@@ -0,0 +1,169 @@
+import { Component, h, Prop, State } from '@stencil/core';
+
+interface Column {
+  header: string;
+  type: 'string' | 'number';
+}
+
+@Component({
+  tag: 'ip-table',
+  styleUrl: 'table.scss',
+  shadow: true,
+})
+export class IpTable {
+  @Prop() columns: string;
+  @Prop() rows: string;
+
+  @State() parsedThead: Column[] = [];
+  @State() parsedTbody: string[][] = [];
+  @State() sortedColumn: number = null;
+  @State() isAscending = true;
+
+  componentWillLoad() {
+    this.parseColumns(this.columns);
+    this.parseRows(this.rows);
+  }
+
+  parseColumns(newValue: string) {
+    try {
+      const parsedColumns: Column[] = JSON.parse(newValue || '[]');
+
+      if (
+        Array.isArray(parsedColumns) &&
+        parsedColumns.every(
+          (item) =>
+            typeof item === 'object' &&
+            item !== null &&
+            'header' in item &&
+            'type' in item &&
+            (item.type === 'string' || item.type === 'number'),
+        )
+      ) {
+        this.parsedThead = parsedColumns;
+      } else {
+        console.error(
+          "Invalid columns structure. Expected an array of objects with 'header' and 'type' properties.",
+        );
+      }
+    } catch (error) {
+      console.error('Invalid columns:', error);
+    }
```

I think we should throw exception if error occurred instead of log